components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/listener/WorkflowAuditLogger.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.workflow.mgt.listener;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.workflow.mgt.bean.Parameter;
+import org.wso2.carbon.identity.workflow.mgt.bean.Workflow;
+import org.wso2.carbon.identity.workflow.mgt.bean.WorkflowRequest;
+import org.wso2.carbon.identity.workflow.mgt.exception.WorkflowException;
+
+import java.util.List;
+
+public class WorkflowAuditLogger extends AbstractWorkflowListener {
+
+    private static final Log AUDIT_LOG = CarbonConstants.AUDIT_LOG;
+    private static final String AUDIT_MESSAGE = "Initiator : %s | Action : %s | Data : { %s } | Result " +
+            ":  %s ";
+    private static final String AUDIT_SUCCESS = "Success";
+
+
+    /**
+     * Trigger after deleting the request
+     *
+     * @param workflowRequest
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostDeleteWorkflowRequest(WorkflowRequest workflowRequest) throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Request ID" + "\" : \"" + workflowRequest.getRequestId()
+                + "\"";
+        AUDIT_LOG.info(String.format(AUDIT_MESSAGE, loggedInUser, "Remove workflow request", auditData,
+                AUDIT_SUCCESS));
+    }
+
+    /**
+     * Trigger after delete the workflow
+     *
+     * @param workflow
+     * @throws WorkflowException
+     */
+    @Override
+    public void doPostDeleteWorkflow(Workflow workflow) throws WorkflowException {
+        String loggedInUser = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isBlank(loggedInUser)) {
+            loggedInUser = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        String auditData = "\"" + "Workflow ID" + "\" : \"" + workflow.getWorkflowId()
+                + "\"";
```

Comment:
Thanks for pointing this, fixed with https://github.com/wso2/carbon-identity/pull/1675